docs/release-strategy.md

```diff
@@ -0,0 +1,68 @@
+# InstructLab CLI Release Strategy
+
+This document discusses the release strategy and processes for the
+`instructlab` Python package built from the
+<https://github.com/instructlab/instructlab> git repository.
+
+## Versioning Scheme
+
+Releases use a `X.Y.Z` numbering scheme.
+
+X-stream release are for major releases. At this stage in the project a major release has not been cut and we expect each release (2-3 weeks) to be a new Y-stream.
+
+Backwards compatibility will exist within a Y-stream for example within 0.1.8 and 0.1.9 and between Y-stream releases for update purposes. For example between 0.1.9 and 0.2.0.
+
+Z-stream releases are meant for critical bug and documentation fixes. Backward compatibility is guaranteed between Z-stream releases. Z-stream releases are cut as maintainers see fit.
+
+## Schedule
+
+The project currently operates on a time-based release schedule.
+The goal is to release a new Y-stream every 2-3 weeks though the time span remains flexible based on the discretion of the maintainers team.
+
+The cadence for major releases starting from 1.0 and onward will be determined as the project matures.
+
+A schedule will be updated in a markdown file on the <https://github.com/instructlab/instructlab> GitHub repository.
+
+## Release Tracking
+
+Release planning is tracked via milestones in GitHub. Milestones on GitHub will exist for the next two releases.
+
+PRs and issues associated with the next two milestones will be prioritized for review and merging into the main branch of instructlab.
+
+## Git Branches and Tags
+
+Every `X.Y` release stream gets a new branch.
+
+Each release, `X.Y.Z`, exists as a tag named `vX.Y.Z` on the `release-vX.Y` branch.
+
+## Release Branch Maintenance
+
+Maintenance efforts are primarily focused on the `main` branch.
+Occasionally, critical bug fixes may be backported to the most recent
+`X.Y` branch for a Z-stream bug-fix release. Bug fixes in general are not
+actively backported to release branches.
+
+## Release Mechanics
+
+The following are the steps for how a release gets cut.
+
+1. Maintainers determine a commit on the main branch that will serve as the basis for the next release.
+    - For a Z-stream release skip this step.
+2. For a Y-Stream release: create a new branch in the format `release-vX.Y`.
+    - For a Z-Stream release skip this step.
+3. Create a commit for the release.
+    - The subject of the commit will be `vX.Y.Z release` and there will be no body in the commit.
+4. Open up a PR with the commit for the release.
+5. Maintainers Merge PR.
+    - The PR will require 2 maintainer approvals.
```

Comment:
I don't think you can create a completely empty commit. Is the intent to put tagging of a release through a review process? I think it is, but let me know if I'm wrong ...

An alternative to having that review is to create the branch and then create a PR on that release branch that updates a ChangeLog file or something like that with a version and some changelog text (could just be a link to the github milestone? tbd ...)

Reply:
the way I understand it was this:

so everytime we do a new Y-stream:
1. new branch off main, tag X.Y.0 and stable

new Z stream:
1. backport commits to existing release branch
3. new tag for Z stream off latest commit to release branch
4. move stable tag to the same commit (should match new tag in step 2)

Reply:
@cdoern and I will be doing a Z-stream release today, I will update here based on our experience

Reply:
This is what we did for `0.14.1`
1. `git checkout --track upstream/release-0.14`
2. `git tag -a v0.14.1` (create new release tag)
3. `git tag -fa stable` (move `stable` tag)
4. `git push upstream --tags --force` (force needed for moving `stable` tag)
5. Draft new release in the GitHub UI **with the existing new tag** (`v0.14.1` in this case)